controllers/namespace-delete/controller.go

```diff
@@ -0,0 +1,91 @@
+package nsdelete
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/storageos/api-manager/internal/pkg/storageos"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	// MaxConcurrentReconciles is the maximum number of concurrent Reconciles which can be run.
+	MaxConcurrentReconciles = 5
```

Comment:
How about making this configurable as a flag or env var?
We may need to adjust it based on the cluster namespaces. Default 5 should be good.